### PR TITLE
Remove implicit `Fix` embedding in tests

### LINKED ISSUE #1652: remove implicit `Fix` embedding in tests
In some [test helpers](https://github.com/quasar-analytics/quasar/blob/master/sql/src/test/scala/quasar/sql/helpers.scala#L73) we have

``` scala
implicit def toFix[F[_]](unFixed: F[Fix[F]]): Fix[F] = Fix(unFixed)
```

Remove this implicit as it doesn't make tests any simpler, and instead makes them more complicated because they compile magically even when you forget to `embed`.

This will require improving our test helpers to easily and readably embed into `Fix` in tests. One possible solution is to approach this as suggested in the [Best Practices](https://github.com/slamdata/matryoshka/wiki/Best%5B%E2%80%A0%5D-Practices) wiki.

### PR DESCRIPTION
**Pull request by alissapajer**
Closes #1652.

### CODE REVIEW

File: sql/src/test/scala/quasar/sql/helpers.scala [hunk 1]
```diff
@@ -53,31 +53,26 @@ trait CompilerHelpers extends TermLogicalPlanMatchers {
...
     fullCompile(query).toEither must beRight(equalToPlan(expected))
 
-  def read(file: String): Fix[LogicalPlan] =
+  def read(file: String): Fix[LP] =
     lpf.read(sandboxCurrent(posixCodec.parsePath(Some(_), Some(_), κ(None), κ(None))(file).get).get)
 
-  // TODO: This type and implicit are a failed experiment and should be removed,
-  //       but they infect the compiler tests.
-  type FLP = Fix[LogicalPlan]
-  implicit def toFix[F[_]](unFixed: F[Fix[F]]): Fix[F] = Fix(unFixed)
```
Comment (alissapajer) [inline sql/src/test/scala/quasar/sql/helpers.scala]: This implicit and type aliases were removed. The rest of the changes are the result of deleting these two lines.

The implicit meant that you could forget to `.embed` and the code would "magically" compile. It's better practice to instead provide helper functions to aid in easy embedding.

File: frontend/src/main/scala/quasar/frontend/logicalplan/LogicalPlanR.scala [hunk 2]
```diff
@@ -45,6 +45,12 @@ class LogicalPlanR[T[_[_]]: Recursive: Corecursive] {
   def constant(data: Data) = lp.constant[T[LP]](data).embed
   def invoke[N <: Nat](func: GenericFunc[N], values: Func.Input[T[LP], N]) =
     Invoke(func, values).embed
+  def invoke1(func: GenericFunc[nat._1], v1: T[LP]) =
```
Comment (alissapajer) [inline frontend/src/main/scala/quasar/frontend/logicalplan/LogicalPlanR.scala]: We utilize the following functions to help create embedded `LogicalPlan`s.

**Comment by alissapajer** [comment]
The build should pass once #1754 is merged.

**Comment by wemrysi** [comment]
LGTM, will merge once #1754 gets in and travis approves. I cancelled the build of the changelog commit, just to get it out of the queue.

**Comment by alissapajer** [comment]
Sounds good. And thanks, I should have skipped ci on the changelog commit!

**Pull request merged**